File: 7-reverse-integer/reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long long rev = 0;
        bool neg = false;
        if (x < 0){
            neg = true;
            x = abs(x);
        }
        while(x>0){
            int rem = x%10;
            rev = rev*10 + rem;

            x/=10;
        }
        if (neg) rev = -rev;
        if (rev > INT_MAX || rev < INT_MIN){
            return 0;
        }
        return rev;
    }
};

File: 7-reverse-integer/reverse-integer.go
func reverse(x int) int {
    flag := false
    if x < 0 {
        flag = true
        x *= -1
    }

    var rev []byte
    for x > 0 {
        digit := (byte(x % 10))
        rev = append(rev, digit)
        x /= 10
    }
    fmt.Println(rev)

    for i := range rev {
        x = x * 10 + int(rev[i])
    }

    if x < -2147483648 || x > 2147483647 {
        return 0
    }

    if flag {
        return x * -1
    }
    return x
}